tests/Doctrine/Tests/ORM/Mapping/AbstractMappingDriverTest.php

```diff
@@ -1344,6 +1352,47 @@ public static function loadMetadata(ClassMetadataInfo $metadata)
     }
 }
 
+/**
+ * @Entity
+ * @InheritanceType("SINGLE_TABLE")
+ * @DiscriminatorMap({"cube" = cube::class})
+ * @DiscriminatorColumn(name="discr", length=32, type="string")
+ */
+abstract class Shape
```

Comment:
Should this class be moved out of `AbstractMappingDriverTest`? The class already has more than 1000 LoC and it has to start somewhere. :-)

Reply:
Sure thing, any preference for its namespace?